### PR TITLE
Check the that ed25519 and ed448 are allowed by the security level

### PR DESCRIPTION
**Pull request by kroeckx**
Signature algorithms not using an MD weren't checked that they're
allowed by the security level.

##### Checklist

- [x] tests are added or updated

### CODE REVIEW

**Comment by kroeckx** [comment]
Still need to update the tests

File: ssl/t1_lib.c [hunk 1]
```diff
@@ -1062,6 +1062,30 @@ int tls_check_sigalg_curve(const SSL *s, int curve)
...
+        return 0;
+    if (md != NULL)
+    {
+        /* Security bits: half digest bits */
+        secbits = EVP_MD_size(md) * 4;
+    } else {
+        if (lu->sigalg == TLSEXT_SIGALG_ed25519)
+            secbits = 128;
+        else if (lu->sigalg == TLSEXT_SIGALG_ed448)
+            secbits = 224;
```
Comment (mattcaswell) [inline ssl/t1_lib.c]: Is there a reference for these values?

Comment (kaduk) [reply ssl/t1_lib.c]: https://tools.ietf.org/html/rfc8032#section-8.5

Comment (kroeckx) [reply ssl/t1_lib.c]: Yes, see ecx_meth.c line 21 and 27. I guess I could move the define to some public header.

Comment (mattcaswell) [reply ssl/t1_lib.c]: Thanks. Might be worth including the reference @kaduk gave in a comment here (optionally also in ecx_meth.c).

Comment (romen) [reply ssl/t1_lib.c]: I agree adding the reference on both files is a good addition!

**Comment by kroeckx** [comment]
I've updated the tests and fixed a conflict.

**Comment by mspncp** [comment]
Drive by comment: Typo in commit message: „Check ~the~ that“

**Comment by mattcaswell** [comment]
LGTM, except that travis red cross seems relevant.

**Comment by kroeckx** [comment]
For some reason I commited a test change to the .in file

**Pull request closed without merging**